text/0000-expand-math.md

```diff
@@ -1,26 +1,23 @@
-- Feature Name: Expand `math`
+- Feature Name: Introduce new numerics -- `Rational`, `BigInt`, `BigFloat`, `Complex`
 - Start Date: 2022-02-28
 - RFC PR:
 - Pony Issue:
 
 # Summary
 
-This RFC proposes expanding the standard library's `math` package and the shape it might take to leverage the full extent of Pony. There was prior discussion on [Zulip](https://ponylang.zulipchat.com/#narrow/stream/192795-contribute-to.20Pony/topic/math.20lib) years ago, but I expect some opinions to have changed.
+This RFC proposes the introduction of new numeric types; in particular the addition of a type representing a fractional number (`Rational`), arbitrary precision integer (`BigInt`), arbitrary precision float (`BigFloat`), and complex number (`Complex`).
 
 # Motivation
 
-Currently, `math` includes limited functionality. This should be expanded to include math types, constants, and present a structure with further expansion in mind. An expanded `math` library will allow a unified mathematics among Pony developers. As was discussed during our [2022-03-01 Sync](https://sync-recordings.ponylang.io/r/2022_03_01.m4a), one core principle for including these types in the stdlib is to have a single canonical implementation of them which allow interoperability of numeric types across the Pony ecosystem. Any numeric types introduced by this RFC **must** existing within the current numerical type hierarchy by being compliant with existing numeric traits.
+The primary motivation for adding these types to the stdlib is to have a single canonical implementation of them which allow interoperability of numeric types across the Pony ecosystem.
 
 # Detailed design
 
-The primary goals of this initial expansion are:
-
-1. restructure the `math` package into distinct subpackages; allowing for separation of concerns over continuing to build a monolithic `math` package
-2. provide common `math` data types; for example, `BigInt`, `Rational`, and `Complex`
+I propose we add the aforementioned numeric types into `builtin` so they exist alongside the other standard numeric types. These introduced numeric types **must** existing within the current numeric type hierarchy by being compliant with existing numeric traits.
```

Comment:
For me, the preference would be to keep `builtin` as minimal as possible, so I'd prefer to put these new types in a new package (or series of packages), unless there is a strong motivation for it to go in `builtin`.

Pretty much all of the existing public types in `builtin` have hard requirements forcing them to be there, with a reason like one of the following:
- they are used by core language constructs (`None`, string literals, numeric literals, etc)
- they need to use raw pointers (`Array`, `String`, etc)
- they use `compile_intrinsic` for one or more function definitions
- they are part of the `Env`, which is passed to the `Main` actor on entry
- they are used by one of the types that meet the above criteria

I don't think these new types have any hard requirements forcing them to go in `builtin`, so I believe they shouldn't be there.

Reply:
We can discuss this further on Sync and I will update the RFC according to our conversation. I have no particular need for these to be added to `builtin` so no objection to an agreed upon other location such as a newly created `numerics` or adding these to `math` (as was the RFC state prior to my latest changes).